src/ring_theory/dedekind_domain.lean

```diff
@@ -30,6 +30,9 @@ giving three equivalent definitions (TODO: and shows that they are equivalent).
 The definitions that involve a field of fractions choose a canonical field of fractions,
 but are independent of that choice. The `..._iff` lemmas express this independence.
 
+Often, definitions assume that Dedekind domains are not fields. We found it more practical
+to add a `(h : ¬ is_field A)` assumption whenever this is explicitly needed.
```

Comment:
I had always assumed that there was universal agreement that Dedekind domains are not fields. Turns out that there is actually literature to back up your proposed definition. It seems that wiki and Matsumura both allow fields to be Dedekind domains.

Reply:
My instinct was the same as Johan's but I agree that most of the literature seems not to assume that Dedekind domains are non-fields! So I'm definitely happy.